File: javaScalaCheck/src/main/java/properties/Dates.java
package properties;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.time.temporal.ChronoUnit;
import java.time.temporal.TemporalAccessor;
import java.time.temporal.TemporalUnit;
import java.util.Calendar;
import java.util.Date;

public class Dates {
	static String pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS";
	static DateFormat dateFormat = new SimpleDateFormat(pattern);

	public static String nextDay(String s) throws ParseException {
		Calendar c = Calendar.getInstance();
		c.setTime(fromString(s));
		c.add(Calendar.DAY_OF_YEAR, 1);
		return toString(c.getTime());
	}

	public static String nextDayJava8(String s) throws ParseException {
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern);
		LocalDateTime localDate = LocalDateTime.parse(s, formatter);
		LocalDateTime date2 = localDate.plus(1, ChronoUnit.DAYS);
		return date2.format(formatter);
	}

	public static String toString(Date d) {
		return dateFormat.format(d);
	}

	public static Date fromString(String s) throws ParseException {
		return dateFormat.parse(s);
	}

}

File: kotlin/src/main/kotlin/properties/Dates.kt
package properties

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.time.temporal.ChronoUnit
import java.util.*

object Dates {
    internal val pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS"
    internal val dateFormat: DateFormat = SimpleDateFormat(pattern)

    fun toString(d: Date) = dateFormat.format(d)!!
    fun fromString(s: String) = dateFormat.parse(s)!!

    fun nextDay(s: String): String {
        val c = Calendar.getInstance()
        c.time = fromString(s)
        c.add(Calendar.DAY_OF_YEAR, 1)
        return toString(c.time)
    }

    fun nextDayJava8(s: String): String {
        val formatter = DateTimeFormatter.ofPattern(pattern)
        val localDate = LocalDateTime.parse(s, formatter)
        val date2 = localDate.plus(1, ChronoUnit.DAYS)
        return date2.format(formatter)
    }

}
